carbonmark-api/src/utils/helpers/utils.ts

```diff
@@ -219,3 +220,24 @@ export const isMatchingCmsProject = (
   { registry, projectId }: IsMatchingCmsProjectArgs,
   project: CarbonProject
 ) => project?.registryProjectId === projectId && project.registry === registry;
+
+/**
+ * Converts a String into an ActivityType without breaking typescript checks
+ */
+export const stringToActivityType = (str: string): ActivityType | undefined => {
+  for (const [key, value] of Object.entries(ActivityType)) {
+    if (key == str) {
+      return value;
+    }
+  }
+};
+/**
+ * Converts an array of Strings into an array of ActivityType without breaking typescript checks
+ * FIXME: this method will silently filter out strings that do not relate to an actually activity type
+ * potentially misleading people using the API
```

Comment:
We can fix this by using a Type.Literal and Type.Union in the model for activity if you know what the possible types are:

```ts
Type.Union([
  Type.Literal("Type 1"),
  Type.Literal("Type 2"),
]);
```

Reply:
Great tip! 
Done in b9b5dd00f